pkg/kbld/image/preresolved.go

```diff
@@ -17,6 +17,13 @@ func NewPreresolvedImage(url string, origins []ctlconf.Origin) PreresolvedImage
 }
 
 func (i PreresolvedImage) URL() (string, []ctlconf.Origin, error) {
+	for _, origin := range i.origins {
+		if origin.Preresolved != nil && origin.Preresolved.URL == i.url {
+			imageOrigins := copyAndAppendOrigins(i.origins)
+			return i.url, imageOrigins, nil
+		}
+	}
```

Comment:
what if we do a more generic check for any dups (exact match) within metas?

Reply:
Spoke over zoom and it will be simpler to not make a generic solution due to this bug only manifesting for preresolved images